addthis.js

```diff
@@ -1,33 +1,53 @@
+/**
+ * @file
+ * AddThis javascript actions.
+ *
+ * @todo Should async be in here?
+ *   Question if we need to do anything with the async setting. The developer
+ *   might need to take action itself with addthis.init().
+ */
+
 (function ($) {
+  // This load the config in time to run any addthis functionality.
+  addthis_config = Drupal.settings.addthis.addthis_config;
+  addthis_share = Drupal.settings.addthis.addthis_share;
 
   Drupal.behaviors.addthis = {
     attach: function(context, settings) {
 
-      // If addthis settings are provided a display is loaded.
-      if (typeof Drupal.settings.addthis != 'undefined') {
-
-        if (typeof Drupal.settings.addthis.load_type != 'undefined') {
-          if (Drupal.settings.addthis.load_type == 'async') {
-            if (typeof addthis != 'undefined') {
-              addthis.init();
-            }
-          }
-          if (Drupal.settings.addthis.load_type == 'domready') {
-            $.getScript(
-              Drupal.settings.addthis.widget_url,
-              function(data, textStatus) {});
-          }
-          // Trigger ready on ajax attach.
-          if (context != window.document &&
-              typeof window.addthis != 'undefined' &&
-              typeof window.addthis.toolbox == 'function')
-          {
-              window.addthis.toolbox('.addthis_toolbox');
-          }
-        }
+      // Trigger ready on ajax attach.
+      if (context != window.document) {
+        Drupal.behaviors.addthis.ajaxLoad(context, settings);
+      }
+
+    },
+
+    // Load the js library when the dom is ready.
+    loadDomready: function() {
+      // If settings asks for loading the script after the dom is loaded, then
+      // load the script here.
+      if (Drupal.settings.addthis.domready) {
+        $.getScript(Drupal.settings.addthis.widget_url, Drupal.behaviors.addthis.scriptReady);
       }
+    },
 
+    // Called when a ajax request returned.
+    ajaxLoad: function(context, settings) {
+      if (typeof window.addthis != 'undefined' &&
+          typeof window.addthis.toolbox == 'function')
+      {
+          window.addthis.toolbox('.addthis_toolbox');
+      }
```

Comment:
This will never be reached as you execute ```if (context != window.document) {``` in the attach.
If you remove that code from the attach, you can register if addthis is initialized in the Drupal.behaviors.addthis object